master/internal/api_user_intg_test.go

```diff
@@ -925,6 +927,62 @@ func TestPostUserActivity(t *testing.T) {
 	require.Equal(t, 1, activityCount, ctx)
 }
 
+func TestFilterUserByRoleId(t *testing.T) {
+	api, _, ctx := setupAPITest(t, nil)
+	config.GetMasterConfig().Security.AuthZ = config.AuthZConfig{Type: "rbac"}
+	defer func() {
+		config.GetMasterConfig().Security.AuthZ = config.AuthZConfig{Type: "basic"}
+	}()
+	testRole := rbac.Role{
+		ID:   10002,
+		Name: uuid.New().String(),
+	}
+	_, err := db.Bun().NewInsert().Model(&testRole).Exec(ctx)
+	require.NoError(t, err, "failure creating role in setup")
+
+	user0 := model.User{Username: uuid.New().String()}
+	_, err = db.HackAddUser(context.TODO(), &user0)
+	require.NoError(t, err)
+	user1 := model.User{Username: uuid.New().String()}
+	_, err = db.HackAddUser(context.TODO(), &user1)
+	require.NoError(t, err)
+
+	err = rbac.AddRoleAssignments(ctx, nil,
+		[]*rbacv1.UserRoleAssignment{
+			{
+				UserId: int32(user0.ID),
+				RoleAssignment: &rbacv1.RoleAssignment{
+					Role: &rbacv1.Role{
+						RoleId: int32(testRole.ID),
+					},
+					ScopeWorkspaceId: nil,
+				},
+			},
+			{
+				UserId: int32(user1.ID),
+				RoleAssignment: &rbacv1.RoleAssignment{
+					Role: &rbacv1.Role{
+						RoleId: int32(testRole.ID),
+					},
+					ScopeWorkspaceId: ptrs.Ptr(int32(1)),
+				},
+			},
+		},
+	)
+	require.NoError(t, err, "error adding role assignment")
+	resp, err := api.GetUsers(ctx, &apiv1.GetUsersRequest{
+		RoleIdAssignedDirectlyToUser: []int32{int32(testRole.ID)},
+	})
+	require.NoError(t, err)
+	require.Len(t, resp.Users, 1)
+	require.Equal(t, int32(user0.ID), resp.Users[0].Id)
```

Comment:
we only want `user0` and not `user1` because the query filters to global assignments (ref [here](https://github.com/determined-ai/determined/blob/main/master/internal/api_user.go#L187)) right?

Reply:
Exactly